src/entity/RDFPropertyMap.cpp

```diff
@@ -3,8 +3,54 @@
 
 namespace sempr { namespace entity {
 
+// triple iterator for rdfpropertymap
+
+RDFPropertyMapIterator_impl::RDFPropertyMapIterator_impl(
+    const RDFPropertyMap* pmap, std::map<std::string, RDFValue>::const_iterator it)
+    : it_(it), pmap_(pmap)
+{
+}
+
+const Triple& RDFPropertyMapIterator_impl::operator*() const
+{
+    static Triple t;
+    t.subject = pmap_->subject_;
+    t.predicate = "<" + it_->first + ">";
+    t.object = it_->second.toString();
+    t.document = "<" + sempr::baseURI() + pmap_->id() + ">";
+
+    return t;
+}
+
+const Triple* RDFPropertyMapIterator_impl::operator->() const
+{
+    return &(**this);
```

Comment:
And now it gets dirty. You will always return the address to the static **local** variable with this.

Reply:
It works. And it's a static variable. So what? ;D
I agree that this is a bit ugly. Maybe I was way too focused on creating an iterator that I favored the signatures over sensible implementation.